iceoryx_binding_c/CMakeLists.txt

```diff
@@ -52,6 +52,30 @@ set(${PROJECT_NAME}_DIR ${CMAKE_CURRENT_LIST_DIR}/cmake
     FORCE
 )
 
+set(IOX_BINDING_C_STORAGE_SIZE_FILE_PATH "${CMAKE_BINARY_DIR}/generated/iceoryx/include/iceoryx_binding_c/iceoryx_binding_c_storage_sizes.h")
+
+add_executable(iox-binding-c-storage-size-generator source/storage_size_generator.cpp)
+
+target_compile_definitions(iox-binding-c-storage-size-generator
+    PRIVATE
+    IOX_BINDING_C_STORAGE_SIZE_FILE_PATH="${IOX_BINDING_C_STORAGE_SIZE_FILE_PATH}"
+)
+
+target_link_libraries(iox-binding-c-storage-size-generator iceoryx_posh::iceoryx_posh)
+
+# run generator
+add_custom_target(
+    iox-binding-c-storage-size-generator-exec ALL
+    COMMAND ${CMAKE_CURRENT_BINARY_DIR}/iox-binding-c-storage-size-generator
+    BYPRODUCTS
+    ${IOX_BINDING_C_STORAGE_SIZE_FILE_PATH}
+    WORKING_DIRECTORY ${CMAKE_BINARY_DIR}
+    VERBATIM
+    USES_TERMINAL
+)
+
+add_dependencies(iox-binding-c-storage-size-generator-exec iox-binding-c-storage-size-generator)
+
```

Comment:
@mossmaurice @MatthiasKillat @elfenpiff this wasn't as difficult as expected. I remembered that I did something similar for a private project ... hope the CI will be green